docs/proposal/003-dnsendpoint-graduation-to-beta.md

```diff
@@ -0,0 +1,179 @@
+```yaml
+---
+title: "Proposal: Defining a path to Beta for DNSEndpoint API"
+version: v1alpha1
+authors: @ivankatliarchuk, @raffo, @szuecs
+creation-date: 2025-02-09
+status: proposal
+---
+```
+
+# Proposal: Agree on requirements for API DNSEndpoint to graduate to beta
+
+## Summary
+
+The `DNSEndpoint` API in Kubernetes SIGs `external-dns` is currently in alpha. To ensure its stability and readiness for production environments, we propose defining and agreeing upon the necessary requirements for its graduation to beta.
+By defining clear criteria, we aim to ensure stability, usability, and compatibility with the broader Kubernetes ecosystem. On completions of all this items, we should be in the position to graduate `DNSEndpoint` to `v1beta`.
+
+## Motivation
+
+The DNSEndpoint API is a crucial component of the ExternalDNS project, allowing users to manage DNS records dynamically.
+Currently, it remains in the alpha stage, limiting its adoption due to potential instability and lack of guaranteed backward compatibility. By advancing to beta, we can provide users with a more reliable API and encourage wider adoption with confidence in its long-term viability and support.
+
+### Goals
+
+- Define the necessary requirements for `DNSEndpoint` API to reach beta status.
+- Improve API stability, usability, and documentation.
+- Improve test coverage, automate documentation creation, and validation mechanisms.
+- Ensure backward compatibility and migration strategies from alpha to beta.
+- Collect and incorporate feedback from existing users to refine the API.
+- Address any identified issues or limitations in the current API design.
+
+### Non-Goals
+
+- This proposal does not cover the graduation of ExternalDNS itself to a stable release.
+- Making `DNSEndpoint` a stable (GA) API at this stage.
+- It does not include implementation details for specific DNS providers.
+- It does not introduce new functionality beyond stabilizing the DNSEndpoint API.
+- Redesigning the API from scratch.
+- Introducing breaking changes that would require significant refactoring for existing users.
+
+## Proposal
+
+The proposal aims to formalize the promotion process by addressing API design, user needs, and implementation details.
+
+To graduate the `DNSEndpoint` API to beta, we propose the following actions:
+
+1. Capture feedback from the community on missing functionality for DNSEndpoint CRD
+   - In a form of Github issue, pin the issue to the project
+   - Link all CRD related issues to it
+2. Refactor `endpoint` folder, move away `api/crd` related stuff to `apis/<apiVersion> folder`
+3. Documentation for API to be generated automatically with test coverage, similar to `docs/flags.md`
+4. API(s) and CRD(s) discoverable. [doc.crds.dev](https://doc.crds.dev/github.com/kubernetes-sigs/external-dns). Example [crosplane](https://doc.crds.dev/github.com/crossplane/crossplane@v0.10.0)
```

Comment:
```suggestion
4. APIs and CRDs discoverable. [doc.crds.dev](https://doc.crds.dev/github.com/kubernetes-sigs/external-dns). Example [crossplane](https://doc.crds.dev/github.com/crossplane/crossplane@v0.10.0)
```
On this specific point, it worked in the past:
https://doc.crds.dev/github.com/kubernetes-sigs/external-dns@v0.7.6
I'm not sure what has caused this in recent versions.